File: MP.Core/Models/Room.cs
﻿using System;
using System.Collections.Generic;
using System.Text;

namespace MP.Core.Models
{
    public class Room
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public int Capacity { get; set; }
        public int TicketValue { get; set; }
        public int CinemaId { get; set; }
        public Cinema Cinema { get; set; }
    }
}

File: MP.Web/ClientApp/src/models/room.ts
import ICinema from "./cinema";

enum RoomFields {
    NAME = "name",
    CAPACITY = "capacity",
    TICKET_VALUE = "ticketValue",
    CINEMA = "cinema"
}

export default interface IRoom {
    id?: number,
    [RoomFields.NAME]: string,
    [RoomFields.CAPACITY]: number,
    [RoomFields.TICKET_VALUE]: number,
    [RoomFields.CINEMA]: ICinema
}